lib/puppet/defaults.rb

```diff
@@ -854,11 +854,23 @@ def self.default_diffargs
         This is distinct from the certificate authority's CRL."
     },
     :certificate_revocation => {
-        :default  => true,
-        :type     => :boolean,
-        :desc     => "Whether certificate revocation should be supported by downloading a
-          Certificate Revocation List (CRL)
-          to all clients.  If enabled, CA chaining will almost definitely not work.",
+        :default  => 'chain',
```

Comment:
I'm curious why `:chain` ? I'm reading this PR as `true` and `:chain` are synonyms, and to fulfill the purpose of the PR all we strictly need is a new value `:leaf`. What's the direction behind `:chain` ?